pyro/autohelp/ast_visitor.py

```diff
@@ -0,0 +1,246 @@
+                decorator=node.decorator.func
+            )
+        elif (
+            isinstance(node.decorator, libcst.Attribute)
+            and node.decorator.attr.value == "listen"
+        ):
+            self.found_errors.append(Actions.DECORATOR_LISTEN_NOT_CALLED)
+            self.updates[node] = lambda node: node.with_changes(
+                decorator=libcst.Call(node.decorator)
+            )
+
+    def visit_FunctionDef_params(self, node: libcst.FunctionDef) -> Optional[bool]:
+        if not node.params or not node.decorators:
+            return False
+        first_param: libcst.Param = node.params.params[0]
+        if not first_param.name.value == "self":
+            return False
+        # its self and a top level method, so check for a bot decorator
+        # needs a decorator with `command` in the last section and to be called
+        visitor = _FindCommandDecorator()
+        node.visit(visitor)
+        if visitor.is_command_dec:
+            self.found_errors.append(Actions.USING_SELF_ON_BOT_COMMAND)
+
+            def update(node: libcst.FunctionDef):
+                params = node.params.with_changes(params=node.params.params[1:])
+                return node.with_changes(params=params)
+
+            self.updates[node] = update
+
+
+class ClientIsNotBot(BaseHelpTransformer):
+
+    BOT_CLASSES = ("Bot", "InteractionBot")
+    def visit_Assign(self, node: libcst.Assign):
+        # what
+        if not isinstance(node.value, libcst.Call):
+            return
+        # search for client in the assign
+        found_client = False
+        possible_fix = lambda node: node
+        for target in node.targets:
+            if hasattr(target, "attr"):
+                continue
+            if target.target.value == "client":
+                found_client = True
+
+                def possible_fix(node: libcst.Assign):
+                    targets = list([t for t in node.targets])
+                    pos = targets.index(target)
+
+                    targets.pop(pos)
+
+                    targets.insert(
+                        pos,
+                        libcst.AssignTarget(
+                            target=target.target.with_changes(value="bot")
+                        ),
+                    )
+                    return node.with_changes(targets=targets)
+
+                break
+
+        if not found_client:
+            return
+        # search for bot in the class assigment
+        found_bot = False
+
+        # get the last call
+        func = node.value.func
+        if isinstance(func, libcst.Name):
+            if func.value in self.BOT_CLASSES:
+                found_bot = True
+
+        elif isinstance(func, libcst.Attribute):
+            if func.attr.value in self.BOT_CLASSES:
+                found_bot = True
+
+        if not found_bot:
+            return
+
+        self.found_errors.append(Actions.CLIENT_IS_NOT_BOT)
+        return
+        self.updates[node] = possible_fix
```

Comment:
Unreachable code

Reply:
The fix is broken, but might reimplement soon

Reply:
Reimplemented. 
![image](https://user-images.githubusercontent.com/71233171/152733282-9e2d8100-85cb-4aeb-8c46-063641858c46.png)

(renames all client variables to bot)

Reply:
https://github.com/MenuDocs/Pyro/pull/58/commits/b279ad144799ae08e56012dfd674568d6500d145